p2p/src/swarm/mod.rs

```diff
@@ -340,10 +362,10 @@ where
         loop {
             tokio::select! {
                 event = self.rx_swarm.recv().fuse() => {
-                    self.on_swarm_control_event(event).await?;
+                    self.on_swarm_control_event(event).await.map_fatal_err()?;
                 }
                 event = self.handle.poll_next() => match event {
-                    Ok(event) => self.on_network_event(event).await?,
+                    Ok(event) => self.on_network_event(event).await.map_fatal_err()?,
                     Err(e) => {
                         log::error!("failed to read network event: {:?}", e);
                         return Err(e);
```

Comment:
Unrelated to this PR but I noticed that the `run` function returns a `Result<(), Error>` but the `Ok` variant is never returned as far as I can see. It would be more precise to express it in the type with something like `Result<Void, Error>` (which is basically equivalent to just `Error`). Not sure if the `!` type is stable yet, but if it is, that would be even better.

Reply:
Yeah, I'll fix that in a future PR because it also needs to be fixed in all P2P event loops.